ui/perfherder/helpers.js

```diff
@@ -0,0 +1,334 @@
+import chunk from 'lodash/chunk';
+
+import { tValueCareMin, tValueConfidence, phTimeRanges } from './constants';
+import { getApiUrl, createQueryParams } from '../helpers/url';
+// TODO move getData into a the global helpers file if using for everything
+import { getData } from '../intermittent-failures/helpers';
+import PerfSeriesModel from '../models/perfSeries';
+
+export const calcPercentOf = function calcPercentOf(a, b) {
+  return b ? 100 * a / b : 0;
+};
+
+export const calcAverage = function calcAverage(values) {
+  if (!values.length) {
+    return 0;
+  }
+
+  return values.reduce((a, b) => a + b, 0) / values.length;
+};
+
+export const getStdDev = function getStandardDeviation(values, avg) {
+  if (values.length < 2) {
+    return undefined;
+  }
+
+  if (!avg) avg = calcAverage(values);
+
+  return Math.sqrt(
+    values.map(v => (v - avg) ** 2).reduce(function (a, b) { return a + b; }) / (values.length - 1));
+};
+
+// If a set has only one value, assume average-ish-plus standard deviation, which
+// will manifest as smaller t-value the less items there are at the group
+// (so quite small for 1 value). This default value is a parameter.
+// C/T mean control/test group (in our case original/new data).
+export const getTTest = function getTTest(valuesC, valuesT, stddev_default_factor) {
+  const lenC = valuesC.length;
+  const lenT = valuesT.length;
+
+  if (!lenC || !lenT) {
+    return 0;
+  }
+
+  const avgC = calcAverage(valuesC);
+  const avgT = calcAverage(valuesT);
+  let stddevC = (lenC > 1 ? getStdDev(valuesC, avgC) : stddev_default_factor * avgC);
+  let stddevT = (lenT > 1 ? getStdDev(valuesT, avgT) : stddev_default_factor * avgT);
+
+  if (lenC === 1) {
+    stddevC = valuesC[0] * stddevT / avgT;
+  } else if (lenT === 1) {
+    stddevT = valuesT[0] * stddevC / avgC;
+  }
+
+  const delta = avgT - avgC;
+  const stdDiffErr = (
+    Math.sqrt(
+      (stddevC * stddevC / lenC) // control-variance / control-size
+        +
+      (stddevT * stddevT / lenT),
+    )
+  );
+
+  return delta / stdDiffErr;
+};
+// TODO many of these are only used in one controller so can likely be moved
+// into the appropriate react component
+
+// eslint-disable-next-line arrow-body-style
```

Comment:
Unless in exceptional circumstances, I'd like to avoid adding `eslint-disable` lines to ~new code

Reply:
I was getting hit with both the arrow-body-style and no-confusing-arrow rules so probably best to make it a function declaration.